src/k8s/cmd/k8s/k8s_join_cluster.go

```diff
@@ -55,10 +55,14 @@ func newJoinClusterCmd(env cmdutil.ExecutionEnvironment) *cobra.Command {
 				opts.name = hostname
 			}
 
-			if opts.address == "" {
-				opts.address = util.NetworkInterfaceAddress()
+			address, err := utils.ParseAddressString(opts.address, config.DefaultPort)
+			if err != nil {
+				cmd.PrintErrf("Error: Failed to parse the address %q.\n\nThe error was: %v\n", opts.address, err)
+				env.Exit(1)
+				return
 			}
-			opts.address = util.CanonicalNetworkAddress(opts.address, config.DefaultPort)
+
+			opts.address = address
```

Comment:
let's not mix what `opts.address` contains. just use `address` instead of `opts.address` for the rest of the function